snakemake_interface_report_plugins/reporter.py

```diff
@@ -27,6 +28,7 @@ def __init__(
         jobs: List[JobRecordInterface],
         settings: ReportSettingsBase,
         workflow_description: str,
+        metadata: MetadataRecordInterface,
```

Comment:
Turn this into a keyword arg with a default of None that is then initialized as an empty dict.

Reply:
Reason: this way, we don't need a breaking change in the interface.